### ISSUE TITLE
Add pairwise distance to Metrics

### ISSUE THREAD

**Issue by pkdogcom** [pytorch/ignite#93]
I think in evaluation of regression task, pairwise distance, especially norm-2 distance, as in `torch.nn.functional.pairwise_distance` is at least as frequently used as MSE, which is actually mostly used as loss rather than evaluation metrics. Therefore, I was wondering if it is worthy of being added to Metrics package as a commonly used metrics.

**Comment by jasonkriss** [comment]
@pkdogcom Do you mean RMSE? I don't think I've seen (mean) l-2 distance used as a regression metric before. If you do mean RMSE, I think that's a great point. I'll add that.

**Comment by jasonkriss** [comment]
@pkdogcom just added RMSE in #92.

**Comment by pkdogcom** [comment]
@jasonkriss I'm working on gaze estimation which evaluates the performance by using euclidean distance between actual and estimated gaze target at the monitor, averaged over all samples. I believe there are other tasks also use euclidean distance rather than RMSE. Let me know if that makes sense.

**Comment by alykhantejani** [comment]
Seems sensible to me - can you send a PR?

**Comment by jasonkriss** [comment]
@pkdogcom Thanks for the explanation! That makes total sense. I just went ahead and added `MeanPairwiseDistance` to #92. Let me know if that doesn't take care of it.
